src/routing/Router.svelte

```diff
@@ -47,25 +47,30 @@
 
   function useComponent(componentToUse, view) {
     return function handle({ params: { product, section, probeName } }) {
+      const storeValue = get(store);
       component = componentToUse;
       // if the product has changed,
       // set it in the store and use store.resetProductDimensions()
       // to initialize.
-      if (product && $store.product !== product) {
+      if (product && storeValue.product !== product) {
         store.setField('product', product);
         store.resetProductDimensions();
       }
       // Issue #355: Update the probe here, whenever the path changes, to ensure
       // that clicks to the back/forward buttons work as expected.
+
       if (probeName) {
-        store.setField('probeName', probeName);
-        if ($probeSet) {
-          let newProbe = $probeSet.find((p) => p.name === probeName);
-          if (productConfig[$store.product].transformProbeForGLAM) {
-            newProbe = productConfig[$store.product].transformProbeForGLAM(newProbe);
+        // The canonical probe info fetch. (PSS)
+        getSearchResults(probeName).then((r) => {
+          let newProbe = { ...r[0], loaded: true };
```

Comment:
Would it make sense to abstract this code into a little util function called something like `getProbeByName`?